asmcomp/linearize.ml

```diff
@@ -108,6 +108,8 @@ let add_branch lbl n =
   else
     discard_dead_code n
 
+let contains_calls = ref false
+
```

Comment:
Am I mistaken or is this reference essentially an argument of the `linear` function below? Would it not be simpler to declare `linear` as:

```OCaml
let linear contains_call i n =
  let rec linear i n = ...
```
or something similar?

Reply:
Yes, what you suggest is better, but makes a huge and irrelevant diff due to indentation, so I wanted to avoid it to make the review easier. I've just made the change in a separate commit.